### DISCUSSION TITLE
reqire a file be included in request

### DISCUSSION THREAD

**Discussion by CyTechNomad** [jakecoffman/crud#23] [Q&A]
I am using the crud library and would like to require a file file or --data for curl be included in the request. Am i just missing it or is there currently no way to enforce that using the crud library?

**Comment by jakecoffman** [comment]
There's a `File` type, I think that's what you're looking for: https://github.com/jakecoffman/crud/blob/20afaa417fad37acaef34453c6b7db1dd452e5d0/field.go#L309-L312

**Reply by CyTechNomad** [reply]
I am trying the following am i missing something
```
Validate: crud.Validate{
     Body: crud.File().Required(),
},
```

**Reply by jakecoffman** [reply]
This should work for you:

```	
Validate: crud.Validate{
	FormData: crud.Object(map[string]crud.Field{
		"file": crud.File().Required(),
	}),
},
```

**Reply by CyTechNomad** [reply]
I am appreciative of the quick reply. attempting this I am still getting into the endpoint logic without providing a file at all. 

what I am attempting:
 `curl --request POST \
  --url url \
  --header 'Content-Type: text/csv'`
the code im using: 
```
Validate: crud.Validate{
	FormData: crud.Object(map[string]crud.Field{
		"file": crud.File().Required(),
	}),
},
```

**Reply by jakecoffman** [reply]
Ah I see, yes then your approach before is the correct one: 

```
Validate: crud.Validate{
     Body: crud.File().Required(),
},
```

Yes there's currently no validation around the body with this library if the body is a file. To do the validation currently we must read the entire body into memory to deserialize it and validate fields. To do that with a file could run the server out of memory, so you'll have to handle that validation yourself.

**Reply by jakecoffman** [reply]
Out of curiosity what type of validation were you hoping for?

**Reply by CyTechNomad** [reply]
Just trying to ensure a file was provided. Im doing more in-depth validation beyond this but was hoping i could reject the request at this level like i can for other things.
